usecases/MinimumWorkingExample/dodo.py

```diff
@@ -374,3 +385,84 @@ def task_upload_to_openbis():
             'targets': [sample_file_path],
             'clean': [clean_targets],
         }
+
+@create_after(executed='extract_metadata_emodul')
+def task_perform_calibration():
+    """Loop over the experiments and store the calibrated E in csv file. Each iteration generates four files
+    viz. calibrated data.csv, displacement data used, load data used and the knowledge graph file containing the calibration details
+    (these are not specific file names)"""
+    # create folder, if it is not there
+    Path(calibrated_data_directory).mkdir(parents=True, exist_ok=True)
+
+    # defining calibration input, setting the values for the priors
+    E_loc = 30  # KN/mm2 (mean)
+    E_scale = 10  # KN/mm2 (std)
+
+    # setting for fast test, defining the list
+    if config['mode'] == 'cheap':
+        list_exp_name = [single_example_name]
+    else:  # go through all files
+        list_exp_name = os.listdir(
+            processed_data_emodulus_directory)
+        list_exp_name = [os.path.splitext(f)[0] for f in list_exp_name]  # split extension
+    for f in list_exp_name:
+        exp_metadata_path = os.path.join(metadata_emodulus_directory, f + '.yaml')
+        with open(exp_metadata_path) as file:
+            data = yaml.safe_load(file)
+        diameter = float(data['diameter'])
+        length = float(data['length'])
+
+        def calibrate_it(path_exp_data=processed_data_emodulus_directory,path_calibrated_data=calibrated_data_directory,
+                         exp_name=f):
+            output = read_exp_data_E_mod(path=path_exp_data, exp_name=exp_name + '.csv',
+                                         length=length, diameter=diameter)
+            E_samples = estimate_youngs_modulus(experimental_data=output,
+                                                calibration_metadata={"E_loc": E_loc, "E_scale": E_scale},
+                                                calibrated_data_path=path_calibrated_data, mode=config['mode'])
+            # store samples as csv
+            np.savetxt(os.path.join(calibrated_data_directory, f + '_calibrated_samples.csv'), E_samples, delimiter=',')
+        # current outputs
+        ## -- output from probeye. dont really need it but for the time being
+        owl_file = Path(calibrated_data_directory, 'calibrationWorkflow' + f + '.csv.owl')
+        displ_list = Path(calibrated_data_directory, 'displacement_list_' + f + '.csv.dat')
+        force_list = Path(calibrated_data_directory, 'force_list_' + f + '.csv.dat')
+        #another_file = Path(calibrated_data_directory, 'joint_samples_compression_test_calibration.dat')
+        ## -- created by the calibration script
+        calibrated_data_file = Path(calibrated_data_directory,f + '_calibrated_samples.csv')
```

Comment:
I am sorry, I made a mistake. I am talking about the file: "joint_samples_compression_test_calibration.dat"
This is the same name for each sample.
My preference is to create a directory for each sample, otherwise this output gets messy.

Reply:
implemented folder wise results.